File: 프로그래머스/2/42584. 주식가격/주식가격.py
# from collections import deque
def solution(prices):
    n = len(prices)
    # enter = deque()
    answer = []
    for i in range(n):
        time = 0
        j = i + 1
        while j < n:
            time += 1
            if prices[i] > prices[j]:
                break
            else:
                j += 1
        answer.append(time)
    return answer

File: 프로그래머스/2/42584. 주식가격/주식가격.java
import java.util.*;
class Solution {
    public int[] solution(int[] prices) {
        int[] answer = new int[prices.length];
        for(int i=0; i<prices.length; i++){
            answer[i] = prices.length - (i+1);
        }
        Deque<Integer> stack = new ArrayDeque<>();
        for(int i=0; i<prices.length; i++){
            while(!stack.isEmpty() && prices[stack.peek()] > prices[i]){
                int idx = stack.pop();
                answer[idx] = i - idx;
            }
            stack.push(i);
        }
        return answer;
    }
}